### PR TITLE
Adding the circles again

### PR DESCRIPTION
**Pull request by Vic-Dim**

### CODE REVIEW

File: homework_04/04_Victor_Dimitrov/circles.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    '''
+    Simple 2D plane point with 2 coordinates
+    '''
+
+    def __init__(self, coord_x, coord_y):
+        assert(isinstance(coord_x, float) and isinstance(coord_y, float))
+
+        self.coord_x = coord_x
+        self.coord_y = coord_y
+
```
Comment (StelianRBG) [inline homework_04/04_Victor_Dimitrov/circles.py:31]: Good idea to solve problem with single letter variable.

File: homework_04/04_Victor_Dimitrov/circles.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        elif centers_distance == added_radiuses:
+            position = RelativePosition.TOUCHING
+
+        elif substracted_radiuses < centers_distance < added_radiuses:
+            position = RelativePosition.INTERSECTING
+
+        else:
+            position = RelativePosition.SAME
+
```
Comment (StelianRBG) [inline homework_04/04_Victor_Dimitrov/circles.py:84]: Good solve of no-else-return

File: homework_04/04_Victor_Dimitrov/tests.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+from circles import Circle, Point, RelativePosition
+
+def test_no_common_points():
+    '''
+    This test checks if 2 initialized circles have no common points'
+    Two circles have no common points if their centers distance is bigger
+    than the sum of their radii
+    '''
+
```
Comment (StelianRBG) [inline homework_04/04_Victor_Dimitrov/tests.py:14]: Comment is too big!

File: homework_04/04_Victor_Dimitrov/tests.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    assert circle_1.circle_position(circle_2) == RelativePosition.TOUCHING
+
+def test_intersecting():
+    '''
+    Intersecting points have more than 1 common point, but not all of them.
+    In this case we check if the centers euclidean distance is smaller than
+    the radii sum, but it's still bigger than the radii difference.
+    '''
+
```
Comment (StelianRBG) [inline homework_04/04_Victor_Dimitrov/tests.py:53]: Comment is too big!

File: homework_04/04_Victor_Dimitrov/tests.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    assert circle_1.circle_position(circle_2) == RelativePosition.INTERSECTING
+
+def test_same():
+    '''
+    We consider two circles to be the same if one circle contains the whole
+    set of points the second circle has. This means the centers' euclidean
+    distance must be smaller than the radii difference
+    '''
+
```
Comment (StelianRBG) [inline homework_04/04_Victor_Dimitrov/tests.py:75]: Comment is too big!

File: homework_04/04_Victor_Dimitrov/tests.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    assert circle_1.circle_position(circle_2) == RelativePosition.NO_COMMON_POINTS
+
+def test_touching():
+    '''
+    Touching circles have exactly only 1 common point, so the way
+    we test this case is by comparing if the centers' distance is
+    equal to the sum of the radii
+    '''
+
```
Comment (StelianRBG) [inline homework_04/04_Victor_Dimitrov/tests.py:36]: Comment is too big!

File: homework_04/04_Victor_Dimitrov/circles.py [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+        self.coord_x = coord_x
+        self.coord_y = coord_y
+
+    def euclidean_distance(self, other):
+        '''
+        It's handy to have an euclidean distance function instead of writing
+        it every time you need the euclidean distance between two points.
+        Source: https://www.cuemath.com/euclidean-distance-formula/
+        '''
+
```
Comment (StelianRBG) [inline homework_04/04_Victor_Dimitrov/circles.py:38]: Comment is too big!

**Review by StelianRBG — APPROVED** [review]
Good solution of homework 04. My only remark is that you have very big comments and you leave a lot of empty lines.

**Review by Ivan-Enchev — COMMENTED** [review]
Everything seems good. Checks pass.

**Pull request merged**